nextcord/message.py

```diff
@@ -962,6 +963,13 @@ def repl(obj):
         result = pattern.sub(repl, self.content)
         return escape_mentions(result)
 
+    @utils.cached_slot_property('_cs_channel')
+    def channel(self) -> MessageableChannel:
+        if self.flags.ephemeral and isinstance(self._channel, DMChannel):
+            self._channel = self._state.get_channel(self._channel.id)
+            self.guild = self._channel.guild
```

Comment:
guild would now only be accessible if you used .channel first?

Reply:
i will change this